File: algorithms/0628.maximum-product-of-three-numbers/solution.go
func maximumProduct(nums []int) int {
	sort.Slice(nums, func(i, j int) bool {
		return nums[i] < nums[j]
	})

	l := len(nums)
	a := nums[l-1] * nums[l-2] * nums[l-3]
	b := nums[0] * nums[1] * nums[l-1]

	if a > b {
		return a
	} else {
		return b
	}
}

File: algorithms/0628.maximum-product-of-three-numbers/solution.rs
impl Solution {
    pub fn maximum_product(nums: Vec<i32>) -> i32 {
        let mut nums = nums;
        nums.sort_unstable();

        let len = nums.len();
        let a = nums[len - 1] * nums[len - 2] * nums[len - 3];
        let b = nums[0] * nums[1] * nums[len - 1];

        if a > b {
            a
        } else {
            b
        }
    }
}
